### PR TITLE
Roadmap 2024 and Recap for 2023

### PR DESCRIPTION
**Pull request by holgerd77**
This PR adds the recap section for 2023 with bullet points for achievements together with notes on the deviations. Ethers and Portal Network still missing, to be updated by @ricmoo and @ScottyPoi .

I've also prepared a 2024 roadmap document, tracks are somewhat adjusted to reflect new focus point shifts, and I have added updated descriptions for the tracks.

From my side we can merge here and then submit other PRs towards this. 🙂

### CODE REVIEW

**Review by acolytec3 — COMMENTED** [review]
Spoke about this on our team call today but I think we should have something in here about a reimagined browser client.  @g11tech and I are going to have a talk and try and put together something concrete in the next day or two.

File: roadmap/EF2024.md [hunk 1]
```diff
@@ -0,0 +1,200 @@
...
+
+### Ultralight (Portal Network) & Light Clients
+
+With the [Ultralight](https://github.com/ethereumjs/ultralight) client and networking library stack the JavaScript team maintains one of the three leading client implementations for the Portal Network, a protocol for a set of p2p networks that provide lightweight user access to protocol data and solutions for the challenges that L1 and L2 Ethereum clients face.  Ultralight implements a Node.js client serving the Portal-JSONRPC specs, as well a React-based client that runs in a Browser or on a Mobile device.  The Ultralight repository also includes an Browser based interface for exploring and interacting with locally running JSON-RPC clients of any kind.
```
Comment (gabrocheleau) [inline roadmap/EF2024.md:33]:
```suggestion
With the [Ultralight](https://github.com/ethereumjs/ultralight) client and networking library stack the JavaScript team maintains one of the three leading client implementations for the Portal Network, a protocol for a set of p2p networks that provide lightweight user access to protocol data and solutions for the challenges that L1 and L2 Ethereum clients face. Ultralight implements a Node.js client serving the Portal-JSONRPC specs, as well a React-based client that runs in a Browser or on a Mobile device. The Ultralight repository also includes an Browser based interface for exploring and interacting with locally running JSON-RPC clients of any kind.
```

File: roadmap/EF2024.md [hunk 2]
```diff
@@ -0,0 +1,200 @@
...
+
+Development of the [Portal Network](https://github.com/ethereum/portal-network-specs) is ongoing, with Javascript team members taking leading roles in the research and development of current and future Portal Network specifications.  Ultralight is one of three Portal Network client implementations involved in [Portal-Hive](http://github.com/ethereum/portal-hive) interoperability testing, with Javascript team members contributing to the Hive test suites as well.
```
Comment (gabrocheleau) [inline roadmap/EF2024.md:35]:
```suggestion
Development of the [Portal Network](https://github.com/ethereum/portal-network-specs) is ongoing, with Javascript team members taking leading roles in the research and development of current and future Portal Network specifications. Ultralight is one of three Portal Network client implementations involved in [Portal-Hive](http://github.com/ethereum/portal-hive) interoperability testing, with Javascript team members contributing to the Hive test suites as well.  
```

File: roadmap/EF2024.md [hunk 3]
```diff
@@ -0,0 +1,200 @@
...
+
+Development of the [Portal Network](https://github.com/ethereum/portal-network-specs) is ongoing, with Javascript team members taking leading roles in the research and development of current and future Portal Network specifications.  Ultralight is one of three Portal Network client implementations involved in [Portal-Hive](http://github.com/ethereum/portal-hive) interoperability testing, with Javascript team members contributing to the Hive test suites as well.  
+
+The PortalNetwork teams are currently overseeing the testing and deployment of the first networks.  2024 we will focus on observing the behavior and performance of these early networks and finding solutions issues that arise, as well as developing the Portal Network project towards full State access and other goals.  The Ultralight team will also continue to develop the Ultralight client and networking stack, while addressing the security and technical challenges of running a client in a browser environment as well as on mobile devices.
```
Comment (gabrocheleau) [inline roadmap/EF2024.md:37]:
```suggestion
The PortalNetwork teams are currently overseeing the testing and deployment of the first networks. In 2024 we will focus on observing the behavior and performance of these early networks and finding solutions issues that arise, as well as developing the Portal Network project towards full State access and other goals. The Ultralight team will also continue to develop the Ultralight client and networking stack, while addressing the security and technical challenges of running a client in a browser environment as well as on mobile devices.
```

File: roadmap/EF2024.md [hunk 4]
```diff
@@ -0,0 +1,200 @@
...
+
+We have tested an early-on prototype connecting the Lodestar consensus client to the Portal beacon network with Ultralight, which might lead to a new light client architecture for the beacon chain. We will continue and broaden experiments here as a focus for 2024. In parallel other Portal-based network applications are continued to evolved and tested regarding production feasibility.
```
Comment (gabrocheleau) [inline roadmap/EF2024.md:39]: I don't understand this last sentence:
```
In parallel other Portal-based network applications are continued to evolved and tested regarding production feasibility.
```

File: roadmap/EF2024.md [hunk 5]
```diff
@@ -0,0 +1,200 @@
...
+
+### Ethers.js
+
+The [Ethers.js](https://github.com/ethers-io/ethers.js) library is one of the most popular tooling libraries used to and interact with the Ethereum network. Primarily maintained by [Richard Moore](https://github.com/ricmoo) the library is downloaded by far over 1 million users and developers each month and is among the top 500 packages on NPM (by dependents). To further develop and maintain the library is therefore crucial to support the large community which has grown around the library over the years.
```
Comment (gabrocheleau) [inline roadmap/EF2024.md:43]:
```suggestion
The [Ethers.js](https://github.com/ethers-io/ethers.js) library is one of the most popular tooling libraries used to interact with the Ethereum network. Primarily maintained by [Richard Moore](https://github.com/ricmoo) the library is downloaded by far over 1 million users and developers each month and is among the top 500 packages on NPM (by dependents). To further develop and maintain the library is therefore crucial to support the large community which has grown around the library over the years.
```

File: roadmap/EF2024.md [hunk 6]
```diff
@@ -0,0 +1,200 @@
...
+
+### Ethers.js
+
+- Largely development of [Ethers](https://github.com/ethers-io/ethers.js) v7
+  - Account Abstraction [EIP-4337](https://eips.ethereum.org/EIPS/eip-4337) as a first-class citizen, refactoring Runner
+  - Large focus on backwards compatibility; the migration from v5 to v6 was painful for many, but necessary, but with v7 the API changes must remain minimal
+- Testing import times; using c4 create a test suite to test performance of common function and general import time
+- Help major projects with any migration
+- Updating existing tools, such as the Playground and release the Toolbox
+- Burn down older issues
```
Comment (gabrocheleau) [inline roadmap/EF2024.md:92]:
```suggestion
- Largely development of [Ethers](https://github.com/ethers-io/ethers.js) v7
- Account Abstraction [EIP-4337](https://eips.ethereum.org/EIPS/eip-4337) as a first-class citizen, refactoring Runner
- Large focus on backwards compatibility; the migration from v5 to v6 was painful for many, but necessary, but with v7 the API changes must remain minimal
- Testing import times; using c4 create a test suite to test performance of common function and general import time
- Help major projects with any migration
- Updating existing tools, such as the Playground and release the Toolbox
- Burn down older issues
```

File: roadmap/EF2024.md [hunk 7]
```diff
@@ -0,0 +1,200 @@
...
+
+---
+
+## Q2
+
+### Hardforks & TypeScript/Browser DevEx
+
+- 2-3 funktional EIP implementations for anticipated feature HF
```
Comment (gabrocheleau) [inline roadmap/EF2024.md:100]:
```suggestion
- 2-3 functional EIP implementations for anticipated feature HF
```

File: roadmap/EF2024.md [hunk 8]
```diff
@@ -0,0 +1,200 @@
...
+- General support for Ethers and expanding support team
+- Burn down older issues
+
+---
+
+## Q3
+
+### Hardforks & TypeScript/Browser DevEx
+
+- Implementation finalizations and testing for 2-3 funktional EIP implementations for anticipated feature HF
```
Comment (gabrocheleau) [inline roadmap/EF2024.md:140]:
```suggestion
- Implementation finalizations and testing for 2-3 functional EIP implementations for anticipated feature HF
```

File: roadmap/EF2024.md [hunk 9]
```diff
@@ -0,0 +1,200 @@
...
+### Verkle Roadmap Integration
+
+- Functional/isolated implementation of logic related to the Merkle->Verkle transition
+- Sync with future testnet iterations and participate in Merkle->Verkle transition testing
+
+### Ultralight (Portal Network) & Light Clients
+
+  - R&D towards portalnetwork support of Merkle->Verkle transition
+  - Maintain Ultralight public bootnodes and bridge nodes
+  - Observe, analyze, and debug early portal network behavior
```
Comment (gabrocheleau) [inline roadmap/EF2024.md:160]:
```suggestion
- R&D towards portalnetwork support of Merkle->Verkle transition
- Maintain Ultralight public bootnodes and bridge nodes
- Observe, analyze, and debug early portal network behavior
```

File: roadmap/EF2024.md [hunk 10]
```diff
@@ -0,0 +1,200 @@
...
+### Verkle Roadmap Integration
+
+- Participate in Verkle transition testnets and be "verkle-ready".
+
+### Ultralight (Portal Network) & Light Clients
+
+- Implementation and testing of Verkle solutions in parallel with L1 testnets
+- Functional integration of portal clients with light clients
+- Maintain Ultralight public bootnodes and bridge nodes
+- Update ultralight client and networking stack with latest portal-network-specs changes
```
Comment (gabrocheleau) [inline roadmap/EF2024.md:193]:
```suggestion
- Update Ultralight client and networking stack with latest portal-network-specs changes
```

File: roadmap/README.md [hunk 11]
```diff
@@ -10,8 +10,144 @@ list for an overview of the initial roadmaps from the respective years.
...
+
+- Launch portalnetwork v1.0.0 with history/light client update/state network functionality
+- Improve Ultralight documentation and quickstart guide.
+- Maintain Ultralight public bootnodes and bridge nodes.
+- Update and improve Ultralight browser tools, demos, and documentation.
+- History Network Reference Production Integration Partnership (based on Q3 analysis)
+
+### Ethers.js
+
+- This will be (likely) almost entirely focused on v7 features, refactoring and
```
Comment (gabrocheleau) [inline roadmap/README.md:39]:
```suggestion
- This will be (likely) almost entirely focused on v7 features, refactoring and 
```
and [...] ?

**Review by scorbajio — APPROVED** [review]
ACK

**Comment by holgerd77** [comment]
Oh, Amir is not yet added as a team member here (thanks for review/approval), will trigger this to change, but this needs to go though EF DevOps.

For now I will admin merge due to time constraints.

**Pull request merged**